helix-term/src/commands.rs

```diff
@@ -5125,9 +5125,12 @@ fn indent(cx: &mut Context) {
 
     let transaction = Transaction::change(
         doc.text(),
-        lines.into_iter().map(|line| {
+        lines.into_iter().filter_map(|line| {
+            if doc.text().get_line(line).unwrap().as_str().unwrap().trim() == "" {
```

Comment:
Not sure what's the best approach here, initially I was checking that the length of the repo is equal to 1 (new line) but that could break for "\r\n". On the other hand, the current approach will skip also lines that already have for example a few spaces (which might be desired).

Reply:
You can use `doc.text().line(line)` to avoid an unwrap.

Also, https://docs.rs/ropey/1.3.2/ropey/struct.RopeSlice.html#method.as_str
> For large slices this method will typically fail and return None because large slices usually cross chunk boundaries in the rope.

You'll have to do `.into::<Cow<str>>()` I think.

Reply:
What about in languages where whitespace indentation is meaningful?

If we only check for empty lines then this could be simplified `.line(...).len_chars() == 0`

Reply:
> What about in languages where whitespace indentation is meaningful?

Are there any languages though that care about indentation on blank lines? Afaik even for example python will have zero indentation on an empty line. But I might be oblivious to some others.

Reply:
I don't know of any language where indentation on a blank line matters.

Reply:
Let's use `Cow::from(doc.text().line(line)).trim().is_empty()` to avoid the unwrap.

Reply:
Updated, thanks for suggestion.

Reply:
I think this is better although it is longer, it does not need any allocations.
```rust
            let chunks = doc.text().line(line).chunks();
            if chunks.map(str::trim).all(str::is_empty) {          
```

Reply:
Nice, thanks for suggestion, updated.

Reply:
Seems to still use the old suggestion though